### ISSUE TITLE
Core Description Review

### ISSUE THREAD

**Issue by arouinfar** [phetsims/circuit-construction-kit-dc#233]
# CCK: DC 1.5 Core Description Review Checklist

Goals of the Checklist

1. Meet WCAG: Ensure WCAG for Core Description requirements are being met, as often as possible.  
2. Meet PhET Standard: Provide best possible user experience given resource constraints.  
3. Reduce the time that QA spends reviewing sim for items that could be caught earlier in design.  
4. Provide Quick Reference as complement to the Core Description Guide

Audience (who will use this?)

1. Designer who created the Core Description to ensure they haven’t forgotten anything.  
2. Additional content expert to ensure language is pedagogically correct and relevant.  
3. An external reviewer (preferably someone with either content knowledge or core description knowledge) to ensure quality of description is there.
   
## General Instructions

As you work through this list, be sure to keep notes to share with others at weekly Core Description meeting (idea of “continuous learning”).

Test using the simulation [A11y View](https://bayes.colorado.edu/launchpad/chipper/wrappers/a11y-view/?sim=circuit-construction-kit-dc&brand=phet&ea&debugger), as well as testing with screen-reader software to confirm the content is accessible and accurate.

- [Tip sheet VoiceOver](https://docs.google.com/document/d/1qz0Dm2lA67tRhgw1GaHVeOSnldBoMj7AT5UE_UaXz1U/edit)  
- [Tip sheet NVDA](https://docs.google.com/document/d/1pgfyEER7ZlpJlXSwvSCbNBuoCa5oOexc7QvTuFZu-Mo/edit)  
- [Tip sheet JAWS](https://docs.google.com/document/d/1aggemqGsb2CdR7PxgLG50kOg4ZwBPM2M3eI3okyZHJ8/edit)

As you work through the list, keep these 'general good-practice' reminders in the forefront.

**Goals:** Verify Descriptions/A11y View captures the spirit of the design doc with reasonable accuracy
- Does each description component sound natural? (Read it aloud to yourself)  
- Does it need more grammar/syntax? Less?
- Are descriptions (names, help text, summaries, etc.) pedagogically relevant? Decorative content and sometimes spatial placement are not critical for all sims.
- Anything the team has already decided is out of scope, for example, is [documented](#determined-to-be-out-of-scope-for-core-description) to inform the reviewer in advance

## Sim-Specific Instructions

- [Design Doc](https://docs.google.com/document/d/1l9ABgqq4zgsU0f3nW4naylJ9fXGkX4RhBcLNrg0lsLE/edit?tab=t.0#heading=h.amblyaukylkf)
- [.yaml file](https://github.com/phetsims/circuit-construction-kit-common/blob/main/circuit-construction-kit-common-strings_en.yaml)
- The Intro Screen is a subset of the Lab Screen. There's no need to review both screens, so please focus on the Lab Screen.
- There are lots of custom alt input interactions in this sim, please refer to the Keyboard Shortcuts dialog.
- Making connections and moving measurement probes uses a ComboBox interaction. To visually see the ComboBox, run the sim with the `dev` query parameter.

### Determined to be out of scope for Core Description

* Deferred description issues: https://github.com/phetsims/circuit-construction-kit-common/issues?q=is%3Aissue%20state%3Aopen%20label%3Adesign%3Adescription%20label%3Apriority%3A5-deferred 

## Reviewer Checklist

**DATE OF REVIEW COMPLETION:** 2/17/2026

**REVIEWER:** Matthew Moore

### PDOM Structure & Navigation

- [x] Ensure tab order makes sense (limit bouncing, similar items grouped)  
- [x] Headings are used only when necessary, limit sub-levels  
- [x] Components in each heading section make sense for the context of the sim (e.g., Play Area vs. Control Area)

### Screen Summary (review for each screen)

- [x] Overview is succinct, static (if possible), general  
- [x] Describes current screen and includes most important components  
- [x] Current details is at a high level (generic) and supports different presets  
  - Example: PhET-iO hide/show, and/or don’t mention things that could be hidden  
- [x] Interaction hint is screen-specific and supports different presets (e.g., PhET-iO hide/show)

### Home Screen

- [x] Each screen has accessible names  
- [x] Help text gives “playful” invitation to explore the screen

### Accessible Names

- [x] Unique  
- [x] Clear  
- [x] accessibleRoleDescription not repeated in name  
- [x] Generally, title case with no punctuation. Exceptions: Visual name is not capitalized ("Move mass 1"), and critical visual details are added to name using commas ("Carbon Monoxide⁩, ⁨CO⁩, ⁨Linear⁩").

### Accessible Help Text

- [x] Implicitly guides users, as needed. Adds interest or gives a bit of "why try this?"  
- [x] Starts with verb when appropriate, consistent with role  
- [x] Always true for any state
- [x] Consider benefits of a single instance (or combined wording) of help text for similar interactions (e.g., Beaker Solution Controls in Molarity: "Solute amount and solution volume allow changes to chosen solution.")
- [x] Generally, sentence case with punctuation. Exceptions: An accessible name can appear with title casing when used as a parameter in a dynamic help text description. Section headings referred to in help text can also appear like a proper name with title casing. (e.g., "Reset Balloon to start position and an unchanged state.")

### Custom Objects

- [x] Confirm custom objects have a designed accessible role description (to override default: `web-application`)  
- [x] Role and name and/or help text are consistent (e.g., when role is movable, use ‘move’ not ‘drag’)
- [x] Help Text for custom objects can get complicated, especially for multi-step interaction. Guidance needs to be accurate at each step, and to keep the help text simple it can be helpful to refer to the Keyboard Shorcuts dialog. (e.g., "Look for grab button to play. Once grabbed, use keyboard shortcuts to move balloon. Space to release.") (SEE [DESIGN GUIDE](https://docs.google.com/document/d/1kCivjmuXiMzrFkYUigZFgDkssoEWGW_-OaXDk9myV00/edit?tab=t.0#heading=h.rj5etgrq1nf7) FOR DETAILS)

### Visual Content Descriptions

- [x] Non-interactive information is described in static or dynamic state descriptions (e.g., on-screen text, pop ups are described using accessible paragraph)

### Responsive Descriptions

If present, review responses given when pedagogically important events or state changes occur.

- [x] \[Object Responses\] match format: Phrase fragments, no sentence casing, no period at end  
- [x] Numerics in Object Responses are formatted to read naturally  
  - e.g., For ordered pairs in a coordinate grid, no ( ) are used  
  - e.g., Decimals are rounded to 2 or 3 decimal places to match visible accuracy  
- [x] \[Context Responses\] match format: Full phrases, sentence casing, with punctuation  
- [x] Context Responses are limited to pedagogically important and/or simple changes in the sim to confirm a user’s action or describe the impact of the action  
- [x] In design doc: Object/Context responses are labeled correctly  
- [x] Review order of content responses  
  - e.g., changing info is prioritized as first-read info, (“2.74, \-1.89, on primary parabola” rather than “on primary parabola, 2.74, \-1.89”)  
  - e.g., object responses come before context responses  
- [x] Provide suggestions for any obvious and simple responses you feel are missing from the sim

### **Preferences & Keyboard Help Dialog**

- [x] Verify the auto-generated Keyboard Help Dialog visuals and accessible content are clear and correct  
- [x] Verify Sim-specific controls in Preferences have accessible text

### **Capitalization for Core Description**

- [x] accessibleName: Title case, no punctuation. Example: “Concentration Probe". Mentioned exceptions above.
- [x] accessibleHelpText: Sentence case, with punctuation. Example: “Move spheres, closer or further from each other.” 
- [x] accessibleObjectResponse: generally, phrase fragments, requiring no sentence casing, and no period at end. Example: "4 billion kilograms"  
- [x] accessibleContextResponse: full phrases, sentence casing, with punctuation. Example: “Current parabola saved to grid.”  
- [x] accessibleParagraph: contains full sentences, with sentence case and punctuation  
- [x] accessibleListNode: Strive for consistency (stick with what sounds natural and either all phrases or all sentences with all the same punctuation)

### **Component-Specific Criteria (recent areas of concern/confusion)**

NOTE: This section lists common problem areas for components noticed during sim design and are recognized to be redundant. See the [Core Description Design Guide](https://docs.google.com/document/d/1kCivjmuXiMzrFkYUigZFgDkssoEWGW_-OaXDk9myV00/edit?tab=t.0#heading=h.rj5etgrq1nf7) for full details and components not listed here.

#### Checkboxes

- [x] Ensure help text always reads true (e.g., “Show or hide…” “Explore with or without…”)  
- [x] Include an Accessible Context Response to give user context for the consequence of their actions

#### Accessible Paragraphs or Accessible Headings

- [x] Verify that additional descriptions are clear, concise, and necessary.  
- [x] Verify that components are grouped under related headings (nested headings in buttons don’t look like headings in A11y view right now. ERASE WHEN FIXED)  
- [x] Paragraphs exist for text inside accordion boxes and pop-ups

**Comment by Matthew-Moore240** [comment]
## Home Screen
### Original Design




### Reviewer's Suggestion




### Reason for Change


### Issues


### General Notes for Home Screen






## Intro Screen
### Original Design




### Reviewer's Suggestion




### Reason for Change

 


### Issues







### General Notes for Intro Screen


## Lab Screen
### Original Design




### Reviewer's Suggestion



### Reason for Change
 


### Issues





### General Notes for Lab Screen
